SimTracker/SiPhase2Digitizer/plugins/Pixel3DDigitizerAlgorithm.cc

```diff
@@ -343,11 +336,12 @@ std::vector<digitizerUtility::SignalPoint> Pixel3DDigitizerAlgorithm::drift(
 // Signal is already "induced" (actually electrons transported to the
 // n-column) at the electrode. Just collecting and adding-up all pixel
 // signal and linking it to the simulated energy deposit (hit)
-void Pixel3DDigitizerAlgorithm::induce_signal(const PSimHit& hit,
-                                              const size_t hitIndex,
-                                              const uint32_t tofBin,
-                                              const Phase2TrackerGeomDetUnit* pixdet,
-                                              const std::vector<digitizerUtility::SignalPoint>& collection_points) {
+void Pixel3DDigitizerAlgorithm::induce_signal3D(std::vector<PSimHit>::const_iterator inputBegin,
```

Comment:
why is this changed of name? I thought this was meant to override the method of the mother class:

https://github.com/cms-sw/cmssw/blob/44c3f64c0f5ce5c08f31fb622bc94de026c84263/SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizerAlgorithm.h#L193-L197

in this way this method is never called.
@duartej for your information.

Reply:
I changed the name because I got this msg
"Pixel3DDigitizerAlgorithm.h:55:8: error: 'void Pixel3DDigitizerAlgorithm::induce_signal()' marked 'override', but does not override"
but I guess that was the wrong resolution of the problem, I should have just made it `virtual`

Reply:
> I should have just made it `virtual`

I fail to see how that would solve the issue. The argument list of the derived class method should match the one of the mother class to allow the shadowing of the virtual mother class method from the concrete implementation.

Reply:
So making it virtual gives 
```
hidden overloaded virtual function 'Phase2TrackerDigitizerAlgorithm::induce_signal' declared here: different number of parameters (7 vs 6)
```